src/audio/asrc/asrc_ipc4.c

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019-2023 Intel Corporation. All rights reserved.
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/ipc-config.h>
+#include <rtos/alloc.h>
+#include <rtos/init.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/trace/trace.h>
+#include <sof/common.h>
+#include <rtos/string.h>
+#include <ipc/stream.h>
+#include <user/trace.h>
+#include <errno.h>
+#include "asrc.h"
+
+/* 66b4402d-b468-42f2-81a7-b37121863dd4 */
+DECLARE_SOF_RT_UUID("asrc", asrc_uuid, 0x66b4402d, 0xb468, 0x42f2,
+		    0x81, 0xa7, 0xb3, 0x71, 0x21, 0x86, 0x3d, 0xd4);
+
+DECLARE_TR_CTX(asrc_tr, SOF_UUID(asrc_uuid), LOG_LEVEL_INFO);
+
+int asrc_dai_configure_timestamp(struct comp_data *cd)
+{
+	if (cd->dai_dev) {
```

Comment:
I think all these would look simpler with
```
if (!cd->dai_dev)
    return -ENODEV;
```
(or whatever error code you prefer)

Reply:
It's called "Early Return".

https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

It reduces indentation and is much easier to read. It's similar to `assert()`.

Reply:
agree, done.